Fix pipelines after GitLab migration

lucas-p-bernardes opened the pull request:
This PR fixes pipelines after GitLab migration, that broke mainly because of permissions and paths.

lucas-p-bernardes committed:
aval-tests: Update AVAL image reference to use dynamic CI/CD variables

Replaced the hardcoded registry URL in aval-docker.Dockerfile with GitLab
CI/CD variables. Previously, it was using the internal GitLab, that now is
deprecated. Also, since it is not possible to pull containers across projects
in gitlab.com, use a token that allows this.

Signed-off-by: Lucas Bernardes <lucas.bernardes@toradex.com>

lucas-p-bernardes committed:
gitlab-ci: Update common pipeline to use dynamic project ID variables

Signed-off-by: Lucas Bernardes <lucas.bernardes@toradex.com>

lucas-p-bernardes committed:
tests/integration: Rotate encryption key of Cloud credential files

Signed-off-by: Lucas Bernardes <lucas.bernardes@toradex.com>

lucasbernardestoradex commented:
pipeline: https://gitlab.com/toradex/rd/torizon-core/torizoncore-builder/-/pipelines/1731759953

lucas-p-bernardes commented:
Pipeline: https://gitlab.com/toradex/rd/torizon-core/torizoncore-builder/-/pipelines/1734984351

lucas-p-bernardes committed:
gitlab-ci: export TCBCMD variable in 'test-torizoncore' job

TCBCMD export was removed from 'test-base' job in commit 886c208,
so it's needed to add this export in 'test-torizoncore' job

Signed-off-by: Lucas Bernardes <lucas.bernardes@toradex.com>

lucas-p-bernardes committed:
tests/integration: Improve error handling for Docker network creation

Modify 'tcb_create_docker_network' function to check if the network already
exists before attempting to create it, if yes a message is logged instead of
returning an error. This prevents failures when the network is already present

Signed-off-by: Lucas Bernardes <lucas.bernardes@toradex.com>

lucas-akira reviewed (approved):
LGTM

lucas-p-bernardes commented:
Hello @rborn-tx, can you take a look at this?

rborn-tx commented on tests/integration/samples/credentials/credentials-noprov.zip.enc:
There's a typo on the commit message: "enconding", but actually, I'd say that we're actually updating the encryption key (not any encoding). Suggestion:

```
tests/integration: Rotate encryption key of Cloud credential files
```

lucas-p-bernardes replied:
It's done, thanks @rborn-tx!!

rborn-tx reviewed (commented):
LGTM. Added a suggestion though.

lucas-p-bernardes commented:
@lucas-akira Can we merge this?
